File: app/src/main/java/rod/bailey/trafficatnsw/util/DateUtils.java
package rod.bailey.trafficatnsw.util;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateUtils {

	public static boolean isYesterday(Date someDate) {
		// If the day after 'someDate' is today, then someDate must have been
		// yesterday
		Calendar cal = Calendar.getInstance();
		cal.setTime(someDate);
		cal.add(Calendar.DAY_OF_MONTH, 1);
		return isToday(cal.getTime());
	}

	public static boolean isToday(Date someDate) {
		Calendar today = Calendar.getInstance();
		today.setTime(new Date());

		int todayDay = today.get(Calendar.DAY_OF_MONTH);
		int todayMonth = today.get(Calendar.MONTH);
		int todayYear = today.get(Calendar.YEAR);

		Calendar some = Calendar.getInstance();
		some.setTime(someDate);
		int someDay = some.get(Calendar.DAY_OF_MONTH);
		int someMonth = some.get(Calendar.MONTH);
		int someYear = some.get(Calendar.YEAR);

		return (todayDay == someDay) && (todayMonth == someMonth)
				&& (todayYear == someYear);
	}

	public static String relativeDateAndTime(Date someDate, boolean capitalize) {

		String formatStr = new String();

		if (isToday(someDate)) {
			formatStr = capitalize ? "'Today at' " : "'today at' ";
		} else if (isYesterday(someDate)) {
			formatStr = capitalize ? "'Yesterday at' " : "'yesterday at' ";
		} else {
			formatStr = formatStr + "EEE d MMM ";
		}
		
		formatStr += "h:mm a";

		SimpleDateFormat sdf = new SimpleDateFormat(formatStr);

		return sdf.format(someDate);
	}
}

File: app/src/main/java/rod/bailey/trafficatnsw/util/DateUtils.kt
package rod.bailey.trafficatnsw.util

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date

object DateUtils {
	fun isYesterday(someDate: Date): Boolean {
		// If the day after 'someDate' is today, then someDate must have been
		// yesterday
		val cal = Calendar.getInstance()
		cal.time = someDate
		cal.add(Calendar.DAY_OF_MONTH, 1)
		return isToday(cal.time)
	}

	fun isToday(someDate: Date): Boolean {
		val today = Calendar.getInstance()
		today.time = Date()
		val todayDay = today.get(Calendar.DAY_OF_MONTH)
		val todayMonth = today.get(Calendar.MONTH)
		val todayYear = today.get(Calendar.YEAR)
		val some = Calendar.getInstance()
		some.time = someDate
		val someDay = some.get(Calendar.DAY_OF_MONTH)
		val someMonth = some.get(Calendar.MONTH)
		val someYear = some.get(Calendar.YEAR)

		return todayDay == someDay && todayMonth == someMonth
			&& todayYear == someYear
	}

	fun relativeDateAndTime(someDate: Date, capitalize: Boolean): String {
		var formatStr = String()

		if (isToday(someDate)) {
			formatStr = if (capitalize) "'Today at' " else "'today at' "
		} else if (isYesterday(someDate)) {
			formatStr = if (capitalize) "'Yesterday at' " else "'yesterday at' "
		} else {
			formatStr = formatStr + "EEE d MMM "
		}

		formatStr += "h:mm a"
		val sdf = SimpleDateFormat(formatStr)

		return sdf.format(someDate)
	}
}
